presto-orc/src/main/java/com/facebook/presto/orc/reader/BooleanStreamReader.java

```diff
@@ -58,9 +59,12 @@
 
     private boolean rowGroupOpen;
 
-    public BooleanStreamReader(StreamDescriptor streamDescriptor)
+    private AggregatedMemoryContext systemMemoryContext;
+
+    public BooleanStreamReader(StreamDescriptor streamDescriptor, AggregatedMemoryContext systemMemoryContext)
```

Comment:
I think stream reader constructors should take a `LocalMemoryContext`, so we shouldn't call `newLocalMemoryContext` in `assureVectorSize` methods. With that the `streamReadersSystemMemoryContext` aggregated memory context in `OrcRecordReader` will have one local memory context per stream reader, and the stream readers will own (be responsible for managing the lifecycle of) those local memory contexts. 

The local memory contexts owned by the stream readers should be closed when they no longer hold onto the buffers. IIRC we were talking about adding `close()` methods to the stream readers to do that, and those `close()` methods should be called from `OrcRecordReader.close()`.